src/test/test_script.py

```diff
@@ -72,16 +301,17 @@ def test_script_push_from_bytes(self):
             self.assertEqual(written, len(data)/2 + len(prefix)/2)
 
     def test_wally_witness_program_from_bytes(self):
-        valid_cases = [('00' * 20, 0),
-                       ('00' * 32, 0),
-                       ('00' * 50, SCRIPT_HASH160),
-                       ('00' * 50, SCRIPT_SHA256)]
+        valid_cases = [('00' * 20, 0, '0014'+'00'*20),
+                       ('00' * 32, 0, '0020'+'00'*32),
+                       ('00' * 50, SCRIPT_HASH160, '0014f71015b29ff2583558877ed16a59e4f8f451daa3'),
+                       ('00' * 50, SCRIPT_SHA256, '0020cc2786e1f9910a9d811400edcddaf7075195f7a16b216dcbefba3bc7c4f2ae51')]
 
         out, out_len = make_cbuffer('00' * 100)
-        for data, flags in valid_cases:
+        for data, flags, exp_program in valid_cases:
             in_, in_len = make_cbuffer(data)
             ret, written = wally_witness_program_from_bytes(in_, in_len, flags, out, out_len)
             self.assertEqual(ret, WALLY_OK)
+            self.assertEqual(out[:(len(exp_program)//2)], unhexlify(exp_program))
```

Comment:
nit: We could possibly compare to out[:written] here which would also test that written is correct

Reply:
Updated while merging, thanks